app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,287 @@
+                          },
+                          validator: (String? value) {
+                            return (value != null &&
+                                    !EmailValidator.validate(value))
+                                ? 'Invalid email address'
+                                : null;
+                          },
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.only(top: 16.0),
+                          child: SizedBox(
+                            width: double.infinity,
+                            child: CupertinoSlider(
+                              value: _duration,
+                              min: 1.0,
+                              max: 365.0,
+                              onChanged: (newDuration) =>
+                                  setState(() => _duration = newDuration),
+                            ),
+                          ),
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 6),
+                          child: ElevatedButton(
+                            onPressed: () => submitForm(Navigator.of(context)),
+                            child: Text('Share ${_duration.round()} days'),
+                          ),
+                        ),
+                        error != null
+                            ? Text(
+                                error!,
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor),
+                              )
+                            : Container(),
+                        success != null
+                            ? Text(
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends StatefulWidget {
+  const _ShareHolderList({
+    required this.shareHolders,
+    required this.refreshList,
+    super.key,
+  });
+
+  final dynamic shareHolders;
+  final Function() refreshList;
+
+  @override
+  State<_ShareHolderList> createState() => _ShareHolderListState();
+}
+
+class _ShareHolderListState extends State<_ShareHolderList> {
+  get api => obtainApiClient()!;
+  String? error;
+  String? success;
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.shareHolders == null) {
+      return Center(
+          child: Column(children: const [
+        CircularProgressIndicator(),
+        Text("Loading shares"),
+      ]));
+    }
+
+    if (widget.shareHolders.length == 0) {
+      return const Text('No share holders');
+    }
+
+    return Container(
```

Comment:
Why do you have a `Container` widget here with only a child and no other properties.
I think we can remove this `Container` widget and just return the column